app/controllers/users_controller.rb

```diff
@@ -0,0 +1,79 @@
+class UsersController < ApplicationController
+  before_action :logged_in_user, only: [:index, :edit, :update, :destroy]
+  before_action :correct_user, only: [:edit, :update]
+  before_action :load_user, only: [:show, :edit, :destroy]
+
+  def current_user? user
+    user == current_user
+  end
+
+  def show
+    flash[:warning] = t "not_found" if @user.nil?
+  end
+
+  def index
+    @users = User.paginate(page:
+      params[:page])
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def destroy
+    @user.destroy
+    flash[:success] = t "user_deleted"
+    redirect_to users_url
```

Comment:
xử lý check destroy nữa nè
```suggestion
    if @user.destroy
       flash[:success] = t "user_deleted"
    else
      flash[:error] = ...
    redirect_to users_url
```